share/github-backup-utils/ghe-backup-settings

```diff
@@ -26,16 +26,51 @@ echo "* Transferring license data ..." 1>&3
 ghe-ssh "$host" -- "sudo cat '$GHE_REMOTE_LICENSE_FILE'" > enterprise.ghl
 
 # Function to backup a secret setting to a file.
-#   backup-secret <description> <file-name> <setting-name>
+#   backup-secret <description> <file-name> <setting-name> 
 backup-secret() {
-  echo "* Transferring $1 ..." 1>&3
-  ghe-ssh "$host" -- ghe-config "$3" > "$2+" || (
-    echo "Warning: $1 not set" >&2
+
+  best_effort=false
+  label=""
+  file=""
+  setting=""
+  count=0
+
+  while [ $# -gt 0 ]; do
+    case "$1" in
+      --best-effort)
+        shift 1
+        best_effort=true
+        ;;
+      *)
+        case $count in
+          0)
+            label=$1
+            ;;
+          1)
+            file=$1
+            ;;
+          2)
+            setting=$1
+            ;;
+          *)
+            >&2 echo "To many arguments"
```

Comment:
```suggestion
            >&2 echo "Too many arguments"
```

Reply:
I applied this, but I fat-finged the "close" button so I had to do it manually 😄